db/migrations/1635852419286-setup.ts

```diff
@@ -0,0 +1,61 @@
+import { MigrationInterface, QueryRunner } from 'typeorm'
+
+export class Setup1635852419286 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+        CREATE TABLE IF NOT EXISTS report_definition
+        (
+            id      SERIAL PRIMARY KEY,
+            type    TEXT    NOT NULL,
+            version INTEGER NOT NULL DEFAULT 1
+        );
+    `)
+
+    await queryRunner.query(`
+        CREATE TABLE IF NOT EXISTS field
+        (
+            id         SERIAL PRIMARY KEY,
+            type       TEXT    NOT NULL,
+            name       TEXT    NOT NULL,
+            required   BOOLEAN NOT NULL DEFAULT false,
+            validation TEXT
```

Comment:
What's stored in this field?

Reply:
This specifies each "field" which we will store data against, the type of field if this field is required in order to "complete" a report and any validation (regex) that might be required.

This is then referenced by the `fieldValue` which we'll store the value of each field along with a version number in order to protect against erroneous overwrites.

It's based on Mark's strawman here:
https://miro.com/app/board/o9J_ltODBV0=/

Reply:
Sorry, should have been more specific! I meant the validation field itself, as in does that contain a regex or a snippet of javascript or something else?

Reply:
Yes, the validation field will just contain regex if/where required